### DISCUSSION TITLE
Early Hints and Webpack

### DISCUSSION THREAD

**Discussion by ghost** [symfony/symfony#51610] [Q&A]
Hello !

I recently discovered the concept of `103 Early Hints` and I really like the fact that Symfony is supporting this functionnality.

I had a basic question about implementing this in Symfony. Indeed, I am fairly new to the world of front-end, but from what I understand, when you are making a serious application, you will use Webpack in order to allow much more complexity in your CSS and JS managment. 

In our good old Twig templates, we are not calling a static file in the public directory, we are calling files from Webpack. So we don't know how these JS and CSS files will look like in our public/build directory after being generated by Webpack.

Maybe I am not familiar enough with `WebLink` or `Webpack` and the answer is obvious, but how can such a situation be handled ?

Have a nice day :)

**Comment by treztreiz** [comment]
I'm late but for people still wondering, it is possible to generate the assets links as follow:

```php

public function __construct(private readonly EntrypointLookupInterface $entrypointLookup)
{}

// ...

$styles = array_map(
    fn(string $href): Link => new Link(href: $href)->withAttribute('as', 'style'),
    $this->entrypointLookup->getCssFiles('your_entrypoint_name')
);

$scripts = array_map(
    fn(string $href): Link => new Link(href: $href)->withAttribute('as', 'script'),
    $this->entrypointLookup->getJavaScriptFiles('your_entrypoint_name')
);

// Don't forget to reset or the files won't be injected in the view
$this->entrypointLookup->reset();

$response = $this->sendEarlyHints([
    ...$styles,
    ...$scripts
]);

// ...
```
